### PR TITLE
doc: restructure directories and add quickstart

### PR DESCRIPTION
**Pull request by Shabirmean**
### Description
- This PR is one of the many to come where we re-arrange the directory structure of the repo
- Currently the sample application is nested inside a folder called `point-of-sale-app` from the root of the repo
- Ideally we want to bring everything inside that folder to the root of the repo and delete this folder
- Also we want to move the [anthos-baremetal-edge-deployment](https://github.com/GoogleCloudPlatform/point-of-sale/tree/main/anthos-baremetal-edge-deployment) ansible scripts to the [anthos-samples](https://github.com/GoogleCloudPlatform/anthos-samples) repository
- As a first step I am moving everything inside the `point-of-sale-app` folder to the top level of the repo whilst this is kept as is
- We do this because there are content that is referencing this directory. Once we identify them and update them, we can delete this folder

### Changes
- Move everything inside the `point-of-sale-app` folder to the root of the repo
- Put all the source code specific directories (_api-server_, _inventory_, _payments_, _ui_, _service-sdk_) into a folder called `src`
- Add a new folder called `docs` at the root which will have all the different guideline docs
- Add a new doc to explain the QuickStart for deploying the sample application to a GKE cluster

### Files to be reviews:
- The many files in this PR are just from moving the files inside `point-of-sale-app` out into the root directory
- So only files that you might have to review are:
   - The cloudbuild yaml files in the `.github/cloudbuild` path
   - The README in the root of the repo
   - The docs file [(`quickstart.md`)](https://github.com/GoogleCloudPlatform/point-of-sale/tree/restructure1/docs) inside the newly created `docs` folder in the root of the repo
   - README changes to [anthos-baremetal-edge-deployment/README.md](https://github.com/GoogleCloudPlatform/point-of-sale/pull/115/files#diff-9d90e9da61e1a89fe18dbcaef0876334f747020b4cbae5cb037dddb71ee94d98)
   - Updates to the [release-please-config.json](https://github.com/GoogleCloudPlatform/point-of-sale/pull/115/files#diff-c55d4dcb68c348b2c06d263819702fbce72eeeb35b662956d02049a5a18fcf2b) file
   - A small addition to the `.gitignore` file
   - Update the `skaffold.yaml` file to have the `src/` prefix before the projects

### CODE REVIEW

**Comment by Shabirmean** [comment]
:zap:  Two deployments have been created for the Point-of-Sale application. You may access and test them at: :zap: </br>
- **MySQL DB** backed deployment: [**34.136.23.171**](http://34.136.23.171)
- **Embedded H2** DB backed deployment: [**35.225.234.157**](http://35.225.234.157)

File: docs/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    export PROJECT_ID="<your-project-id>"
+    gcloud services enable \
+      container.googleapis.com \
+      monitoring.googleapis.com \
+      --project ${PROJECT_ID}
+    ```
+2. **Clone this repository.**
+    ```bash
+    git clone https://github.com/GoogleCloudPlatform/point-of-sale.git
+    cd point-of-sale
```
Comment (bourgeoisor) [inline docs/quickstart.md:27]: Are we intentionally referencing the "old" directory path, in this tutorial?

**Review by bourgeoisor — APPROVED** [review]
This LGTM based on the description of what needs to be reviewed + my comment + a small clarification:
- When you say "moved the PoS app to the root directory" do you mean **copy** (and not move), since the old directory with its content is still present?

File: docs/quickstart.md (hunk 1 above)
Comment (Shabirmean) [reply docs/quickstart.md:27]: This is pointing to the new directory path. So we are cloning the repo and moving into the repo.
We are not moving into the `point-of-sale-app` folder inside the repo which is the old path

**Comment by Shabirmean** [comment]
:zap:  Two deployments have been created for the Point-of-Sale application. You may access and test them at: :zap: </br>
- **MySQL DB** backed deployment: [**34.136.23.171**](http://34.136.23.171)
- **Embedded H2** DB backed deployment: [**35.225.234.157**](http://35.225.234.157)

**Comment by Shabirmean** [comment]
:zap:  Two deployments have been created for the Point-of-Sale application. You may access and test them at: :zap: </br>
- **MySQL DB** backed deployment: [**34.136.23.171**](http://34.136.23.171)
- **Embedded H2** DB backed deployment: [**35.225.234.157**](http://35.225.234.157)

**Comment by Shabirmean** [comment]
:zap:  Two deployments have been created for the Point-of-Sale application. You may access and test them at: :zap: </br>
- **MySQL DB** backed deployment: [**34.136.23.171**](http://34.136.23.171)
- **Embedded H2** DB backed deployment: [**35.225.234.157**](http://35.225.234.157)

**Pull request merged**